src/mpid/ch4/netmod/ofi/ofi_huge.c

```diff
@@ -16,15 +16,10 @@ int MPIDI_OFI_recv_huge_event(int vni, struct fi_cq_tagged_entry *wc, MPIR_Reque
     MPIR_FUNC_ENTER;
 
     bool ready_to_get = false;
-    /* Check that the sender didn't underflow the message by sending less than
-     * the huge message threshold. When striping is enabled underflow occurs if
-     * the sender sends < MPIDI_OFI_STRIPE_CHUNK_SIZE through the huge message protocol
-     * or < MPIDI_OFI_global.stripe_threshold through normal send */
-    if (((wc->len < MPIDI_OFI_STRIPE_CHUNK_SIZE ||
-          (wc->len > MPIDI_OFI_STRIPE_CHUNK_SIZE && wc->len < MPIDI_OFI_global.stripe_threshold)) &&
-         MPIDI_OFI_COMM(rreq->comm).enable_striping) ||
-        (wc->len < MPIDI_OFI_global.max_msg_size && !MPIDI_OFI_COMM(rreq->comm).enable_striping)) {
-        return MPIDI_OFI_recv_event(vni, wc, rreq, MPIDI_OFI_REQUEST(rreq, event_id));
+    if (MPIDI_OFI_COMM(rreq->comm).enable_striping) {
+        MPIR_Assert(wc->len == MPIDI_OFI_STRIPE_CHUNK_SIZE);
+    } else {
+        MPIR_Assert(wc->len == MPIDI_OFI_global.max_msg_size);
```

Comment:
I really don't like having assertions for things like this. MPICH should really be doing a better job of returning errors instead of asserting all the time...

Reply:
If the code is correct, the assertions should never trigger and are okay to remove. Assertions are just a form of defensive programming. And sometimes we add assertions just to make the semantics of the code clear. Of course, we also use assertions to replace error checking due to laziness, mostly for checking `malloc`s. For the last case, I agree with you that we should do proper error checking.

Reply:
Are we protecting against our own mistake or the user's mistake. I read this as making sure the user didn't overflow the buffer, but that's still a good case where returning error messages would be nice.

Reply:
The data overflow case won't affect `wc->len` for the first chunk